aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -124,11 +125,25 @@ module aptos_framework::coin {
         amount: u64,
     }
 
+    #[event]
+    struct Deposit has drop, store {
+        account: address,
+        coin_type: TypeInfo,
+        amount: u64,
+    }
+
     /// Event emitted when some amount of a coin is withdrawn from an account.
     struct WithdrawEvent has drop, store {
         amount: u64,
     }
 
+    #[event]
+    struct Withdraw has drop, store {
+        account: address,
+        coin_type: TypeInfo,
+        amount: u64,
+    }
+
```

Comment:
Do we have a preference for how we emit events with generics or not? I was wondering this yesterday while using it.

Like how come not:

```rust
struct Withdraw<phantom CoinType> has drop, store {
   account: address,
   amount: u64,
}
```

Reply:
I am dumb. I think you are absolutely right.